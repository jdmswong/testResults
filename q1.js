export default (persons) => persons.map(person => person.DateOfBirth).sort();
