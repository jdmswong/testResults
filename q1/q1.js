export default (persons) => {
  console.log("Question 1:");
  return persons.map(person => person.DateOfBirth).sort();
};
