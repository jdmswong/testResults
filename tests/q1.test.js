import q1 from '../q1';

test('accepts empty array', () => {
  const test1 = [];
  expect(q1(test1)).toEqual([]);
});

test('works with sample data', () => {
  const sampleData = [
    { DateOfBirth: '1975-01-01' },
    { DateOfBirth: '1976-01-01' },
    { DateOfBirth: '1976-01-01' },
    { DateOfBirth: '1976-02-01' },
    { DateOfBirth: '1976-01-02' },
  ];
  expect(q1(sampleData)).toMatchSnapshot();
});
