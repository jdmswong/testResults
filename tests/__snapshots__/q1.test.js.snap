// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with sample data 1`] = `
Array [
  "1975-01-01",
  "1976-01-01",
  "1976-01-01",
  "1976-01-02",
  "1976-02-01",
]
`;
